zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/ElasticsearchStorage.java

```diff
@@ -522,4 +499,22 @@ static boolean isIpAddress(String address) {
 
   ElasticsearchStorage() {
   }
+
+  static final BodyConverter<CheckResult> READ_STATUS = new BodyConverter<CheckResult> (){
+    @Override public CheckResult convert(HttpData body) throws IOException {
+      String result = body.toStringUtf8();
```

Comment:
I sort of wanted to untie these parsers from armeria or netty by using `ByteBuffer` but I guess it's ok. In that case, I'd pass in `ByteBuf` instead of `HttpData` since it's more flexible. e.g., a little better to do something like this to minimize copies for the success case.

```java
try (ByteBufInputStream stream = new ByteBufInputStream(buf.slice(), true)) {
  JsonParser status = enterPath(JSON_FACTORY.createParser(result), "status");
  if (status == null) {
    throw new IllegalArgumentException("Health status couldn't be read " + buf.toString(UTF_8));
  }
}
```

Reply:
hmm not completely on board with this as we have so much complex code as it is. The cases we use string are not called often or in performance sensitive places. do you think calls to health check are going to significantly warrant this?

Reply:
I will change this one anyway :P

Reply:
Yeah it's probably not a big deal for these so don't necessarily need to change it (I was a bit more concerned about the index template call but realized it's memoized). I might experiment with the interface some more in another PR, and it definitely wouldn't be release-bocking. I tend to have some OCD about failure cases slowing down success cases ;)

Reply:
ugh. it far from pretty.

Reply:
yeah just thinking about the actual length of health check responses which is is small and how often we end up converting things to byte arrays anyway. I don't really like so much complex code which ends up being allocatey afterall.

Reply:
here's where I got stuck on this btw. One thing to keep in mind is that I barely understand the code now, and I suspect few in zIpkin do anymore. we do need to be careful how crazy netty we get especially in simple paths.

```
    default ByteBuf toByteBuf(HttpData content) {
      if (content instanceof ByteBufHolder) {
        return ((ByteBufHolder) content).content();
      } else {
        // what to do here?
      }
    }
```

Reply:
Just for reference, it'd be `Unpooled.wrappedBuffer(content.array())` but yeah no worries. I agree we should keep the netty in one place as much as possible, have some ideas on achieving both will explore them later.